kolibri/core/assets/src/core-actions.js

```diff
@@ -216,7 +216,7 @@ function getCurrentSession(store) {
   const id = 'current';
   const sessionModel = SessionResource.getModel(id);
   const sessionPromise = sessionModel.fetch({});
-  sessionPromise.then((session) => {
+  return sessionPromise.then((session) => {
```

Comment:
Is this the magical mystical fix I need? Turns out it was indeed a race condition.